src/app/pages/register/register.component.ts

```diff
@@ -218,11 +225,41 @@ export class RegisterComponent {
   }
 
   private completeRegistration() {
-    console.log('Submitting registration...', {
-      bundle: this.bundleForm.value,
-      registration: this.registrationForm.value,
-      terms: this.termsForm.value,
-    });
+    this.isSubmitting = true;
+    this.errorMessage = null;
+
+    const registrationData: RegistrationRequest = {
+      first_name: this.registrationForm.get('firstName')?.value || '',
```

Comment:
Aren't all these fields mandatory? Seems weird to have them falling back to `''`. Can you assume they'll be present with `!` if this happens after validation? Is there an error we need to catch here if they can end up blank?